homeassistant/components/myio/sensor.py

```diff
@@ -0,0 +1,112 @@
+"""Support for myIO sensors."""
+from datetime import timedelta
+import logging
+
+from homeassistant.const import CONF_NAME, TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=3)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a sensor entity from a config_entry."""
+
+    _server_name = slugify(config_entry.data[CONF_NAME])
+    _server_status = hass.states.get(_server_name + ".state").state
+    _server_data = hass.data[_server_name]
+    _sensors = _server_data["sensors"]
+
+    sensor_entities = []
+    if _server_status.startswith("Online"):
```

Comment:
This is already checked before forwarding the entry setup?

Reply:
this checks, if the server goes offline on the fly...

Reply:
The data coordinator already implements this. The `available` properties of entities can be used to reflect that.

Reply:
Ok. I replaced with this solution... if it is better as to You...